vnext/Microsoft.ReactNative/Modules/DeviceInfoModule.cpp

```diff
@@ -0,0 +1,116 @@
+          strongHolder->updateDeviceInfo();
+        }
+      });
+
+  deviceInfoHolder->m_dpiChangedRevoker = displayInfo.DpiChanged(
+      winrt::auto_revoke, [weakHolder = std::weak_ptr(deviceInfoHolder)](const auto &, const auto &) {
+        if (auto strongHolder = weakHolder.lock()) {
+          strongHolder->updateDeviceInfo();
+        }
+      });
+}
+
+void DeviceInfoHolder::notifyChanged() noexcept {
+  if (m_notifyCallback) {
+    m_notifyCallback(getDimensions());
+  }
+}
+
+React::JSValueObject DeviceInfoHolder::GetDimensions(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto holder = propertyBag.Get(winrt::Microsoft::ReactNative::ReactPropertyId<
+                                winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+      L"DeviceInfo", L"DeviceInfoHolder"));
+
+  return (*holder)->getDimensions();
+}
+
+React::JSValueObject DeviceInfoHolder::getDimensions() noexcept {
+  return React::JSValueObject{
+      {"windowPhysicalPixels",
+       React::JSValueObject{{"width", m_windowWidth * m_scale},
+                            {"height", m_windowHeight * m_scale},
+                            {"scale", m_scale},
+                            {"fontScale", m_textScaleFactor},
+                            {"densityDpi", m_dpi}}},
+      {"screenPhysicalPixels",
+       React::JSValueObject{{"width", m_screenWidth},
+                            {"height", m_screenHeight},
+                            {"scale", m_scale},
+                            {"fontScale", m_textScaleFactor},
+                            {"densityDpi", m_dpi}}},
+  };
+}
+
+void DeviceInfoHolder::SetCallback(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag,
+    Mso::Functor<void(React::JSValueObject &&)> &&callback) noexcept {
+  auto holder = propertyBag.Get(winrt::Microsoft::ReactNative::ReactPropertyId<
+                                winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+      L"DeviceInfo", L"DeviceInfoHolder"));
+
+  (*holder)->m_notifyCallback = std::move(callback);
+}
+
+void DeviceInfoHolder::updateDeviceInfo() noexcept {
+  auto const &displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  auto const &window = xaml::Window::Current().CoreWindow();
+  winrt::Windows::UI::ViewManagement::UISettings uiSettings;
+
+  m_windowWidth = window.Bounds().Width;
+  m_windowHeight = window.Bounds().Height;
+  m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
+  m_textScaleFactor = uiSettings.TextScaleFactor();
+  m_dpi = displayInfo.LogicalDpi();
+  m_screenWidth = displayInfo.ScreenWidthInRawPixels();
+  m_screenHeight = displayInfo.ScreenHeightInRawPixels();
+}
+
+void DeviceInfo::getConstants(React::ReactConstantProvider &provider) noexcept {
+  provider.Add(L"Dimensions", std::move(DeviceInfoHolder::GetDimensions(m_context.Properties())));
```

Comment:
>std::move [](start = 30, length = 9)

It is probably do nothing in this context: internally we convert this to call JSValueWriter. I would expect that we do not really 'move' any value here.